### PR TITLE
Disallow importing non-dask estimators from xgboost.dask

### PR DESCRIPTION
**Pull request by gforsyth**
This is mostly a style change, but also avoids a user error (that I have
committed on a few occasions).  Since `XGBRegressor` and `XGBClassifier`
are imported as parent classes for the `dask` estimators, without
defining an `__all__`, autocomplete (or muscle) memory will produce the
following with little prompting:

```
from xgboost.dask import XGBClassifier
```

There's nothing inherently wrong with that, but given that
`XGBClassifier` is not `dask` enabled, it can lead to confusing behavior
until you figure out you should've typed

```
from xgboost.dask import DaskXGBClassifier
```

Another option is to alias import the existing non-dask estimators.

### CODE REVIEW

File: python-package/xgboost/dask.py [hunk 1]
```diff
@@ -71,6 +70,19 @@
 except ImportError:
     TrainReturnT = Dict[str, Any]  # type:ignore
 
+__all__ = [
+    "RabitContext",
+    "DaskDMatrix",
+    "DaskPartitionIter",
+    "DaskDeviceQuantileDMatrix",
+    "DaskScikitLearnBase",
```
Comment (trivialfis) [inline python-package/xgboost/dask.py:78]: I think the base class may be excluded?

File: python-package/xgboost/dask.py [hunk 2]
```diff
@@ -71,6 +70,19 @@
 except ImportError:
     TrainReturnT = Dict[str, Any]  # type:ignore
 
+__all__ = [
+    "RabitContext",
+    "DaskDMatrix",
+    "DaskPartitionIter",
```
Comment (trivialfis) [inline python-package/xgboost/dask.py:76]: The iterator can be excluded.  I should probably rename it ...

File: python-package/xgboost/dask.py [hunk 3]
```diff
@@ -71,6 +70,19 @@
...
+    "DaskDeviceQuantileDMatrix",
+    "DaskScikitLearnBase",
+    "DaskXGBRegressor",
+    "DaskXGBClassifier",
+    "DaskXGBRanker",
+    "DaskXGBRFRegressor",
+    "DaskXGBRFClassifier",
+]
```
Comment (trivialfis) [inline python-package/xgboost/dask.py:84]: Do we need the `train`, `predict` and `inplace_predict`?

Comment (gforsyth) [reply python-package/xgboost/dask.py:84]: yeah, I'll add those

**Comment by gforsyth** [comment]
Ready for another look @trivialfis

**Review by trivialfis — APPROVED** [review]
LGTM, thanks for the PR!

**Pull request merged**